subsys/logging/backends/log_backend_net.c

```diff
@@ -48,12 +46,33 @@ static int line_out(uint8_t *data, size_t length, void *output_ctx)
 {
 	struct log_backend_net_ctx *ctx = (struct log_backend_net_ctx *)output_ctx;
 	int ret = -ENOMEM;
+	struct msghdr msg = { 0 };
+	struct iovec io_vector[2];
+	int pos = 0;
 
 	if (ctx == NULL) {
 		return length;
 	}
 
-	ret = zsock_send(ctx->sock, data, length, ZSOCK_MSG_DONTWAIT);
+#if defined(CONFIG_NET_TCP)
+	char len[sizeof("123456789")];
+
+	if (ctx->is_tcp) {
+		(void)snprintk(len, sizeof(len), "%zu ", length);
+		io_vector[pos].iov_base = (void *)len;
+		io_vector[pos].iov_len = strlen(len);
+		pos++;
+	}
+#endif
```

Comment:
Do we need an else case?

```c
#if defined(CONFIG_NET_TCP)
// ...
#else
	if (ctx->is_tcp) {
		return -ENOTSUP;
	}
#endif
```

Reply:
True, it would make sense here. Will fix.